ragen/env/sudoku/env.py

```
@@ -1,540 +1,348 @@
+        cells_filled = np.count_nonzero(self.current_grid)
+        cells_total = self.grid_size * self.grid_size
+        initial_filled = np.count_nonzero(self.initial_grid)
+        output.append("")
+        output.append(f"Progress: {cells_filled}/{cells_total} cells filled ({initial_filled} initial, {cells_filled - initial_filled} placed)")
+        output.append(f"Steps: {self.num_steps}/{self.max_steps}")
+
+        return "\n".join(output)
 
     def close(self):
+        """Clean up resources."""
         pass
```

Ruff (0.14.7)

Unnecessary `pass` statement

Remove unnecessary `pass`

(PIE790)